sql/data_schema.sql

```diff
@@ -0,0 +1,250 @@
+);
+
+CREATE TABLE IF NOT EXISTS data.sample_sets (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    dataset         integer     NOT NULL REFERENCES data.datasets,
+    "collection"    integer     NOT NULL REFERENCES data.collections,
+    sample_size     integer     NOT NULL,
+    phenotype       varchar(50) NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS data.dataset_versions (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    dataset             integer         NOT NULL REFERENCES data.datasets,
+    dataset_version     varchar(20)     NOT NULL,
+    dataset_description text            NOT NULL,
+    terms               text            NOT NULL,
+    var_call_ref        varchar(50)     DEFAULT NULL,
+    available_from      timestamp       DEFAULT current_timestamp,
+    ref_doi             varchar(100)    DEFAULT NULL,
+    data_contact_name   varchar(100)    DEFAULT NULL,
+    data_contact_link   varchar(100)    DEFAULT NULL,
+    num_variants        integer         DEFAULT NULL,
+    coverage_levels     integer[]       DEFAULT NULL
+);
+
+CREATE TABLE IF NOT EXISTS data.dataset_files (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    dataset_version     integer                 NOT NULL REFERENCES data.dataset_versions,
+    basename            varchar(100)            NOT NULL,
+    uri                 varchar(200)    UNIQUE  NOT NULL,
+    file_size           bigint                  NOT NULL
+);
+
+--------------------------------------------------------------------------------
+-- Variant and coverage data fields
+--
+
+CREATE TABLE IF NOT EXISTS data.variants (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    dataset_version integer REFERENCES data.dataset_versions,
+    variant_type varchar, -- variants go here `"enum": ["DEL", "INS", "DUP", "INV", "CNV", "SNP", "DUP:TANDEM", "DEL:ME", "INS:ME"]`
+    rsid integer,
+    chrom varchar(10),
+    pos integer,
+    ref varchar,
+    alt varchar,
+    site_quality real,
+    orig_alt_alleles varchar[],
+    hom_count integer,
+    allele_freq real,
+    filter_string varchar,
+    variant_id varchar,
+    allele_count integer,
+    allele_num integer,
+    quality_metrics jsonb,
+    vep_annotations jsonb
+);
+
+CREATE TABLE IF NOT EXISTS data.variant_genes (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    variant integer REFERENCES data.variants,
+    gene integer REFERENCES data.genes
+);
+
+CREATE TABLE IF NOT EXISTS data.variant_transcripts (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    variant integer REFERENCES data.variants,
+    transcript integer REFERENCES data.transcripts
+);
+
+CREATE TABLE IF NOT EXISTS data.coverage (
+    id integer PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    dataset_version integer REFERENCES data.dataset_versions,
+    chrom varchar(10),
+    pos integer,
+    mean real,
+    median real,
+    coverage real[]
```

Comment:
I'm worried that we lose the connection to coverage levels (1, 5, 10 and so forth) if it's not documented what this array means. Would be nice with a comment here to explain this array-field.

Reply:
comments added